### PR TITLE
GH-963: Integration tests report cyclic dependencies in fabelhaft snapshot code on Jenkins

### PR DESCRIPTION
**Pull request by lbeurerkellner**
Ticket: https://github.com/eclipse/n4js/issues/963

While fixing the tests, I ran into a edge-case with the headless compiler: Generally, we support the use of a pre-existing target platform location (node_modules folder) in the headless compiler, however the support was not implemented properly. 

In this PR, I specifically addressed an issue where an already-installed version of a package was reinstalled, even if the requested version in the target platform location would be an exact version match. In the future it would be desirable to extend this support to version constraints. As a consequence, the headless compiler will not install missing dependencies, if they are already present in the node_modules folder of the target platform location and they match the requested constraint. To support this properly, we must wait for #940 to be completed.

Another change that is introduced by this PR, is that the headless compiler now adopts an existing target platform definition file (package.json) and does no longer replace it with a default one. This was required, since otherwise the headless compiler cannot properly uninstall dependencies that only exist in the target platform in the wrong version.

### CODE REVIEW

File: plugins/org.eclipse.n4js/src/org/eclipse/n4js/external/LibraryManager.java [hunk 1]
```diff
@@ -345,18 +345,44 @@ private void collectDependencies(ProjectDescription description, Map<String, Str
...
+					final Version versionInstalled = new Version(versionInstalledString);
+					if (versionRequested.compareTo(versionInstalled) == 0) {
+						// already installed
+						continue;
+					}
+				} catch (NumberFormatException e) {
+					logger.logInfo("Checking whether the requested version constraint " + name + "@"
+							+ versionRequestedString
+							+ " matches the installed version of the package, is currently unsupported (use an exact version instead): Package will be re-installed.");
 				}
```
[the remark this answers is not in the document]
Comment (lbeurerkellner) [reply plugins/org.eclipse.n4js/src/org/eclipse/n4js/external/LibraryManager.java]: Very good point, thanks! This really only is a temporary fix and I am looking forward to the proper semver support that will be merged with GH-940.
